mantidproject/mantid
# Fix cancelling saving after attempting to close Workbench

**Description of work.**
When you attempt to close a project with unsaved changes on Workbench, a dialog box asks if you want to save. Clicking yes opens a window for choosing the file name and location. Previously clicking cancel on this window would cause Workbench to close without saving, but now it only closes the Save As dialog and Workbench remains open.

**To test:**
1. Open Workbench and make changes to the project, e.g. load a workspace.
2. Click the X in the corner to close Workbench.
3. When asked if you want to save click Yes.
4. When the Save As dialog opens, check that clicking cancel does not close Workbench.
5. Check that the rest of the saving functionality works as intended, including saving via File -> Save Project As... and File -> Save Project.

Fixes #26515 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
